3 Domain layer/CandidatesEvaluator.Contract/Handlers/ICommandHandler.cs

```diff
@@ -0,0 +1,14 @@
+﻿using CandidateEvaluator.Contract.Commands;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CandidateEvaluator.Contract.Handlers
+{
+    public interface ICommandHandler<T> where T : ICommand
```

Comment:
Make it [contravariant](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/covariance-contravariance/creating-variant-generic-interfaces) with `in` operator like below:
`public interface ICommandHandler<in T> where T : ICommand`